x/mongo/driver/topology/connection_test.go

```diff
@@ -836,30 +836,35 @@ func TestConnection(t *testing.T) {
 		})
 
 		t.Run("pinning", func(t *testing.T) {
-			makeMultipleConnections := func(t *testing.T, numConns int) (*pool, []*Connection) {
+			makeMultipleConnections := func(t *testing.T, numConns int) (*pool, []*Connection, func()) {
 				t.Helper()
 
-				addr := address.Address("")
-				pool, err := newPool(poolConfig{Address: addr})
+				addr := bootstrapConnections(t, 3, func(nc net.Conn) {})
```

Comment:
```suggestion
				addr := bootstrapConnections(t, numConns, func(nc net.Conn) {})
```

And same question about whether the callback needs to call `nc.Close()` after the test completes.

Reply:
Ah, good find!

As far as calling `nc.Close()`, that isn't necessary. As long as all of the returned connections are created by calling `pool.checkOut()` then calling `pool.disconnect()` at the end of the test will close them.